components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/CommonUtil.java

```diff
@@ -328,4 +330,46 @@ public static String extractArchive(File sourceFile, String destination) throws
             throw new APIImportExportException(errorMessage, e);
         }
     }
+
+    /**
+     * This method will be used to generate Endpoint certificates and meta information related to endpoint certs
+     *
+     * @param filePath String of new file path
+     * @param content  String of content to write into the file
+     * @throws IOException If an error occurs when generating new certs and yaml file
+     */
+    public static void generateFiles(String filePath, String content) throws IOException {
+
+        File file = new File(filePath);
+        try (FileOutputStream fos = new FileOutputStream(file)) {
+
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            byte[] bytesArray = content.getBytes();
+
+            fos.write(bytesArray);
+            fos.flush();
+
+        } catch (IOException e) {
+            String errorMessage = "Error while generating meta information of client certificates from path.";
+            throw new IOException(errorMessage, e);
```

Comment:
Wrapping an exception with another exception of the same type can be confusing in a stack trace. Better if we can wrap the IOException in another type of exception such as APIMgtException